src/refcount.rs

```diff
@@ -8,22 +10,35 @@ use crate::private::Sealed;
 /// out more.
 #[derive(Clone)]
 // TODO AtomicBool for disconnected when forcibly stopped ?
-pub struct Strong(pub(crate) Arc<AtomicBool>);
+// The RwLock is there to prevent exposing a temporarily inconsistent strong_count caused by brief
+// Arc::upgrade calls in some `Weak` functions below. If exposed, it could lead to a race condition
+// that can prevent an Actor from being stopped.
+pub struct Strong(pub(crate) Arc<(AtomicBool, DropNotice)>, Arc<RwLock<()>>);
```

Comment:
This unfortunately kinda brings us back to a larger address struct. Maybe it's just a hit we need to take though.

Reply:
I don't see how this can be avoided I'm afraid.
The `RwLock` has to be outside the existing `Arc`, otherwise it would be pointless. 
On 64 bit, this changes the size of `Address` from 16 to 24 bytes, so an 8 byte increase unfortunately.
However, with the extra size comes correctness, so not too bad imho. I'm open to suggestions of course.

Reply:
That's true - it might just be a hit we need to take. I'm happy with this strategy now with the increase, and if someone else in the future wants to reduce the size because they need to for whatever reason, they can go for it and I'll merge that too.